MapleServer2/PacketHandlers/Game/MesoMarketHandler.cs

```diff
@@ -0,0 +1,169 @@
+        ItemNotSold = 0x9,
+        ErrorSearchingRange = 0xA,
+        CantPurchaseOwnMeso = 0xB,
+        InvalidMesoQuantity = 0xF,
+        PriceMustBeWithinRange = 0x10,
+        ItemSoldOut = 0x11,
+    }
+
+    public override void Handle(GameSession session, PacketReader packet)
+    {
+        MesoMarketMode mode = (MesoMarketMode) packet.ReadByte();
+
+        switch (mode)
+        {
+            case MesoMarketMode.Load:
+                HandleLoad(session);
+                break;
+            case MesoMarketMode.CreateListing:
+                HandleCreateListing(session, packet);
+                break;
+            case MesoMarketMode.CancelListing:
+                HandleCancelListing(session, packet);
+                break;
+            case MesoMarketMode.RefreshListings:
+                HandleRefreshListings(session, packet);
+                break;
+            case MesoMarketMode.Purchase:
+                HandlePurchase(session, packet);
+                break;
+            default:
+                IPacketHandler<GameSession>.LogUnknownMode(mode);
+                break;
+        }
+    }
+
+    private static void HandleLoad(GameSession session)
+    {
+        session.Send(MesoMarketPacket.LoadMarket());
+        session.Send(MesoMarketPacket.AccountStats(session.Player.Account.MesoMarketDailyListings, session.Player.Account.MesoMarketMonthlyPurchases));
+
+        List<MesoMarketListing> listings = GameServer.MesoMarketManager.GetListingsByAccountId(session.Player.AccountId);
+        session.Send(MesoMarketPacket.MyListings(listings));
+    }
+
+    private static void HandleCreateListing(GameSession session, PacketReader packet)
+    {
+        long mesos = packet.ReadLong();
+        long price = packet.ReadLong();
+
+        if (session.Player.Account.MesoMarketDailyListings >= int.Parse(ConstantsMetadataStorage.GetConstant("MesoMarketDailyListingsLimit")))
+        {
+            session.Send(MesoMarketPacket.Error((int) MesoMarketError.ReachedListingLimit));
+            return;
+        }
+
+        if (!session.Player.Wallet.Meso.Modify(-mesos))
+        {
+            session.Send(MesoMarketPacket.Error((int) MesoMarketError.NotEnoughMesosToList));
+            return;
+        }
+
+        MesoMarketListing listing = new MesoMarketListing(session.Player, price, mesos);
+
+        session.Player.Account.MesoMarketDailyListings += 1;
+        DatabaseManager.Accounts.Update(session.Player.Account);
+        session.Send(MesoMarketPacket.CreateListing(listing));
+        session.Send(MesoMarketPacket.AccountStats(session.Player.Account.MesoMarketDailyListings, session.Player.Account.MesoMarketMonthlyPurchases));
+    }
+
+    private static void HandleCancelListing(GameSession session, PacketReader packet)
+    {
+        long listingId = packet.ReadLong();
+
+        MesoMarketListing listing = GameServer.MesoMarketManager.GetListingById(listingId);
+        if (listing == null)
```

Comment:
I would recommend you using `is` since it's the new c# standard 
```suggestion
        if (listing is null)
```